tests/FSharp.Core.UnitTests/FSharp.Core/Microsoft.FSharp.Control/Tasks.fs

```diff
@@ -53,6 +53,17 @@ type SmokeTestsForCompilation() =
             t.Wait()
             if t.Result <> 1 then failwith "failed"
 
+    [<Fact>]
+    member _.mergesrc() =
+        task {
+            let! x = Task.FromResult(1)
+            and! y = Task.FromResult(2)
+            return x + y
+        }
+        |> fun t -> 
+            t.Wait()
+            if t.Result <> 3 then failwith "failed"
```

Comment:
Just a couple other tests to verify this works as expected:

- Use `ValueTask`
- Use `Task.Yield()`
- Mix and match some of these in different orders.

Reply:
Done